artmanflow/steps/sources_generation_guest.py

```diff
@@ -0,0 +1,131 @@
+        self.run_command(
+            ['pip3', 'uninstall', '--yes', '-q', 'googleapis-artman'])
+        self.run_command(['pip3', 'install', '-q', '-e',
+                          self._guest.guest_root_subpath('artman')])
+        self.run_command(['rm', '-rf', '/artman'])
+        self.run_command(['rm', '-rf', '/googleapis'])
+        self.run_command(['rm', '-rf', '/toolkit'])
+
+        self._guest.guest_root_subpath('toolkit')
+        artman_config = {
+            'local': {
+                'toolkit': self._guest.guest_root_subpath(repo_names['toolkit'])
+            }
+        }
+        ConfigUtils.dump_config(artman_config, '/root/.artman/config.yaml')
+        # self._docker.run_command(['cat', '/root/.artman/config.yaml'])
+
+    def _run_artman(self, repo_names):
+        apis = self._config['artman_client_yaml_configs']
+
+        for api in apis:
+            path = self._guest.guest_client_yaml_file_path(api['path'])
+            api_name = path.partition('/artman_')[2].partition('.yaml')[0]
+
+            target = api['target'].split(',')
+            if len(target) == 1:
+                target.insert(0, 'generate')
+
+            args = [
+                '--root-dir',
+                self._guest.guest_root_subpath(repo_names['googleapis']),
+                '--config', self._guest.guest_root_subpath(api['path']),
+                '--output-dir',
+                self._guest.guest_output_dir_subpath('artifacts'),
+                '--local'
+            ]
+
+            verbose = ['-v'] if self._config['debug_mode'] else []
+            self.puts("\n> Generating API %s" % api_name)
+            self.run_command(
+                ['python3', '-m', 'artman.cli.main'] + args + verbose + target,
+                self._guest.guest_root_subpath(repo_names['artman']))
+
+    def _archive_artifacts(self):
+        self.run_command(['tar', '-pzcf', 'artifacts.tar.gz',
+                          '--remove-files', '-C', 'artifacts', '.'],
+                         self._guest.guest_output_dir_path())
+        self.change_file_permissions(
+            self._guest.guest_output_dir_subpath('artifacts.tar.gz'))
+
+    # TODO: this is a horrible hack, should be fixed in artman instead
+    def _fix_generator_output(self):
+        art_sub = ['artifacts', 'java']
+        cwd = self._guest.guest_output_dir_subpath(art_sub)
+        self.run_command(
+            ['rm', '-rf', 'proto-google-longrunning-v1'], cwd)
+        self.run_command(
+            ['rm', '-rf', 'grpc-google-longrunning-v1'], cwd)
+
+        lr_dir = 'gapic-google-cloud-longrunning-v1'
+        lr_path = self._guest.guest_output_dir_subpath(art_sub + [lr_dir])
+        if os.path.exists(lr_path):
+            new_lr_dir = 'gapic-google-longrunning-v1'
+            self.run_command(['mv', lr_dir, new_lr_dir], cwd)
+            new_lr_path = self._guest.guest_output_dir_subpath(
+                art_sub + [new_lr_dir])
+            seds = [
```

Comment:
Same thing as before: If we are going to do file contents manipulation, use `re` and `io` and such and do not call out to `sed` on the shell.

Reply:
Same thing as before: seds are removed, it was properly fixed on toolkit side.